core/state-machine/src/ext.rs

```diff
@@ -152,15 +143,31 @@ where
 		self.storage_transaction = None;
 	}
 
+	/// Register the given extension for this instance.
+	pub fn register_extension<E: Any + Sized>(&mut self, ext: E) {
+		println!("REGISTER: {:?} {}", ext.type_id(), std::any::type_name::<E>());
+		self.extensions.register(ext);
+	}
+
+	/// Reset the internal state.
+	pub fn reset(&mut self) {
+		self.storage_transaction = None;
+		self.changes_trie_transaction = None;
+		self.id = rand::random();
+	}
+
+	/// Returns a mutable reference to the overlayed changes.
+	pub fn mut_overlayed_changes(&mut self) -> &mut OverlayedChanges {
```

Comment:
Isn't the convention usually to add `mut` at the end? i.e. `overlayed_changes_mut`?